src/Core/OrganizationFeatures/AuthorizationHandlers/CollectionAccessAuthorizationHandlerBase.cs

```diff
@@ -0,0 +1,99 @@
+﻿using Bit.Core.Context;
+using Bit.Core.Enums;
+using Bit.Core.Repositories;
+using Microsoft.AspNetCore.Authorization;
+
+namespace Bit.Core.OrganizationFeatures.AuthorizationHandlers;
+
+public abstract class CollectionAccessAuthorizationHandlerBase<TRequirement, TResource> : BulkAuthorizationHandler<TRequirement, TResource>
+    where TRequirement : IAuthorizationRequirement
+{
+    private readonly ICurrentContext _currentContext;
+    private readonly ICollectionRepository _collectionRepository;
+    private readonly IOrganizationUserRepository _organizationUserRepository;
+
+    protected CollectionAccessAuthorizationHandlerBase(ICurrentContext currentContext, ICollectionRepository collectionRepository, IOrganizationUserRepository organizationUserRepository)
+    {
+        _currentContext = currentContext;
+        _collectionRepository = collectionRepository;
+        _organizationUserRepository = organizationUserRepository;
+    }
+
+    /// <summary>
+    /// Ensures the acting user is allowed to manage access permissions for the target collections.
+    /// </summary>
+    protected async Task CanManageCollectionAccessAsync(AuthorizationHandlerContext context, IAuthorizationRequirement requirement, IEnumerable<Guid> collectionIds)
+    {
+        if (!_currentContext.UserId.HasValue)
+        {
+            context.Fail();
+            return;
+        }
+
+        var distinctTargetCollectionIds = collectionIds.Distinct().ToList();
+
+        // List of collections the user is performing the operation on
+        var targetCollections = await _collectionRepository.GetManyByManyIdsAsync(distinctTargetCollectionIds);
+
+        // A target collection does not exist, fail the requirement
+        if (targetCollections.Count != distinctTargetCollectionIds.Count)
+        {
+            context.Fail();
+            return;
+        }
+
+        var userOgs = await _currentContext.OrganizationMembershipAsync(_organizationUserRepository, _currentContext.UserId.Value);
```

Comment:
I used `OrganizationMembershipAsync()` as it had a check to ensure the `Organizations` property (of type `List<CurrentContentOrganization>`) in `CurrentContext` is not null. 

- If its not null, meaning it was likely populated from the JWT, it simply returns the existing list of `CurrentContentOrganization`s without a database query. 
- Only if the list is null, does it attempt to query the database to construct the list directly.

This way we can be sure the handler always has a copy of the user's organizations. Admittedly, it seems unlikely for the list to be null if its coming from the JWT, but I assume the logic to build the list from the database in case its missing from the JWT exists for certain scenarios? 

If you think we should instead fail the auth request if the list of `Organizations` is missing from `CurrentContext` I can switch to using the `Organizations` property directly and fail if its null.